lint/linter.py

```diff
@@ -1066,6 +1072,27 @@ def split_match(self, match):
     def process_match(self, m, vv):
         error_type = self.get_error_type(m.error, m.warning)
 
+        # determine a filename for this match
+        filename = m.match.groupdict().get("filename", None)
+        if filename:
+            # ensure that the filename is absolute by basing relative paths on
+            # the working directory
+            filename = os.path.normpath(os.path.join(self.get_working_dir(self.settings), filename))
+
+            try:
+                # if this is a match for a different file we need its contents
+                # for the below checks
+                if not self.filename or not os.path.samefile(filename, self.filename):
```

Comment:
- Avoid `samefile` bc it does a `stat` call
- try/except clause is then too wide
- Can we just do a simple `self.filename != filename`. Do you have concerns about symlinks? Does `realpath` instead be okay?

Reply:
Is this a performance issue for you or more general?

I mainly chose `samefile` because I wanted to make sure that paths like `a/../a/b` are considered equal to `a/b`. But now that we call `normpath` on `filename` a simple comparison should be OK for this.

But on Windows paths are case insensitive and so there is a (small) chance that Sublime opens a file as `C:\file` and we receive it as `c:\file`. I once ran into such a problem with a different plugin. What about this?
```suggestion
                if os.path.normcase(filename) != os.path.normcase(self.filename):
```

I think we can neglect symlinks but if we want to support them, I don't see an advantage of `realpath` over `samefile`.

Reply:
I will switch to `normcase` then. Sublime Text and python (e.g. `abspath`) don't seem to care what the _correct_ path is. For example they open/return lower case drive letters unchanged.